src/opts.c

```diff
@@ -372,7 +372,7 @@ natsOptions_TLSHandshakeFirst(natsOptions *opts)
     LOCK_AND_CHECK_OPTIONS(opts, 0);
 
     opts->tlsHandshakeFirst = true;
-    opts->secure            = true;
+    s = natsOptions_SetSecure(opts, true);
```

Comment:
I would have set "opts->tlsHandshakeFirst = true" only on success of "SetSecure".